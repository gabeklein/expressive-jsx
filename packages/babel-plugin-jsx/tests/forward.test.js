describe("forward", () => {
  transform("className", `
    () => {
      forward: className;

      <div>Hello World</div>
    }
  `);

  transform("will join with existing styles", `
    () => {
      forward: className;
      color: red;

      <div>Hello World</div>
    }
  `);

  transform("will pull className from existing props", `
    (props) => {
      forward: className;

      <div>Hello World</div>
    }
  `);

  transform("will pull className from existing destructure", `
    ({ name }) => {
      forward: className;

      <div>Hello {name}</div>
    }
  `);

  transform("will pull className from declared destructure", `
    (props) => {
      const { name } = props;

      forward: className;

      <div>Hello {name}</div>
    }
  `);

  transform("will not break where ...rest exists", `
    ({ name, ...rest }) => {
      forward: className;

      <div {...rest}>
        Hello {name}
      </div>
    }
  `);

  transform("react ref", `
    ({ active }) => {
      input: {
        disabled = !active;
        color: red;
        forward: ref;
      }
      
      <this>
        <input type="text" />
      </this>
    }
  `)
})