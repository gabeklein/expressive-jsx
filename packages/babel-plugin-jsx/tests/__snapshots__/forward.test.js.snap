// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forward className 1`] = `
  ({ className }) => (
    <div className={className}>
      <div>Hello World</div>
    </div>
  );
`;

exports[`forward react ref 1`] = `
  forwardRef(({ active }, ref) => (
    <input
      disabled={!active}
      ref={ref}
      type="text"
      className="input_70c"
    />
  ));
  
  css(\`
  	.input_70c {
  		color: red;
  	}
  \`);
`;

exports[`forward will join with existing styles 1`] = `
  ({ className }) => (
    <div className={classNames("element_7nq", className)}>
      <div>Hello World</div>
    </div>
  );
  
  css(\`
  	.element_7nq {
  		color: red;
  	}
  \`);
`;

exports[`forward will not break where ...rest exists 1`] = `
  ({ className, name, ...rest }) => (
    <div className={className}>
      <div {...rest}>Hello {name}</div>
    </div>
  );
`;

exports[`forward will pull className from declared destructure 1`] = `
  (props) => {
    const { className, name } = props;
  
    return (
      <div className={className}>
        <div>Hello {name}</div>
      </div>
    );
  };
`;

exports[`forward will pull className from existing destructure 1`] = `
  ({ className, name }) => (
    <div className={className}>
      <div>Hello {name}</div>
    </div>
  );
`;

exports[`forward will pull className from existing props 1`] = `
  (props) => {
    const { className } = props;
  
    return (
      <div className={className}>
        <div>Hello World</div>
      </div>
    );
  };
`;
